native-sql-engine/core/src/main/scala/com/intel/oap/execution/ColumnarBroadcastHashJoinExec.scala

```diff
@@ -141,12 +141,31 @@ case class ColumnarBroadcastHashJoinExec(
     // build check for expr
     if (buildKeyExprs != null) {
       for (expr <- buildKeyExprs) {
-        ColumnarExpressionConverter.replaceWithColumnarExpression(expr)
+        val columnarBuildKeyExpr = ColumnarExpressionConverter.replaceWithColumnarExpression(expr)
+        // Only do the check for the join having condition.
+        if (condition.isDefined) {
```

Comment:
if there's no condition defined, this check will pass even the exprs does not support codegen, seems not enough?

Reply:
Yes, for no condition case, we still need to add the checking in `supportColumnarCodegen` of join operator. Will add it soon.